src/bot/utils.ts

```diff
@@ -246,27 +249,78 @@ export async function waitForConfirmation(
     ],
     ephemeral: true,
     fetchReply: true
+  })) as Message<true>;
+
+  return interactionWaiter({
+    component_type: ComponentType.Button,
+    message,
+    user: interaction.user
   });
+}
 
-  return new Promise((resolve) => {
-    const collector = message.createMessageComponentCollector({
-      max: 1,
-      componentType: ComponentType.Button,
-      filter: (i: MessageComponentInteraction) => i.user.id === interaction.user.id,
-      time: 60_000
-    });
-    collector.once('end', async (interactions, reason) => {
-      const buttonInteraction = interactions.first();
-      if (!buttonInteraction) {
-        if (reason !== 'time') resolve(null);
-        if (message.deletable) await message.delete();
-        await interaction.reply(interactionInfo('Les 60 secondes se sont ecoulées.'));
-        return resolve(null);
+export async function updateProposalsEmbed(
```

Comment:
*pense à utiliser F2*
```suggestion
export async function updateProposalEmbed(
```